expression/builtin_arithmetic_vec.go

```diff
@@ -210,11 +211,124 @@ func (b *builtinArithmeticIntDivideIntSig) vecEvalInt(input *chunk.Chunk, result
 }
 
 func (b *builtinArithmeticPlusIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinArithmeticPlusIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	lh, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(lh)
+
+	rh, err := b.bufAllocator.get(types.ETInt, n)
```

Comment:
Could we reuse the `result` buffer as `rh` here?

Reply:
Yes, thank you for providing me such a fancy idea.  However, I think this will make it more difficult to implement and understand the code. Since the **rh** is allocated from reused slice pool, reusing **result** buffer as **rh** is not producing very much.

Reply:
The code could be changed a little to achieve this goal. Here's the example.

```go
	if err := b.args[1].VecEvalInt(b.ctx, input, result); err != nil {
		return err
	}
    rh := result
```

And all the code below can remain as the same.

Reply:
Brilliant!